client/lib/wp/sync-handler/index.js

```diff
@@ -91,13 +91,14 @@ export class SyncHandler {
 			const localResponseHandler = localRecord => {
 				// let's be optimistic
 				debug( 'localResponseHandler()', localRecord );
-				if ( localRecord ) {
+				if ( localRecord && localRecord.body ) {
 					debug( 'local callback run => %o, params (%o), response (%o)', path, reqParams, localRecord );
 					// try/catch in case cached record does not match expected schema
+					localRecord.body._headers = { requestKey, responseSource: 'local'};
```

Comment:
I'm pretty sure if we should use `_headers` to add local vars since it's part of the server-response.
What do you think to add another key into the body?

```es6
localRecord.body.__sync = {  requestKey, responseSource: 'local' };
```